api/base/serializers.py

```diff
@@ -44,6 +44,48 @@ def format_relationship_links(related_link=None, self_link=None, rel_meta=None,
     return ret
 
 
+class HideIfRetraction(ser.Field):
+    """
+    If node is retracted, this field will return None.
+    """
+    def __init__(self, field, **kwargs):
+        super(HideIfRetraction, self).__init__(**kwargs)
+        self.field = field
+        self.source = field.source
```

Comment:
This is inconsistent with `AllowMissing`, which doesn't set these attributes. Doing this means the following will break:

```python
# Doesn't work
foo = HideIfRetraction(CharField(), required=True)

# Works
foo = HideIfRetraction(CharField(required=True))
```

Honestly, I'm not sure which one is preferred. Glancing at some of the "higher-order" fields in DRF (`DictField`, `ListField`), they don't set these attributes based on the inner field. I would make this and `AllowMissing` consistent with those.